lib/edge_impulse/Kconfig

```diff
@@ -11,7 +11,9 @@ menuconfig EDGE_IMPULSE
 	depends on LIB_CPLUSPLUS
 	depends on NEWLIB_LIBC
 	depends on NEWLIB_LIBC_FLOAT_PRINTF
-	depends on FPU
+	depends on !CONFIG_FP16
+	imply FPU if CPU_HAS_FPU
```

Comment:
Doesn't `imply` mean "try to select if possible"? I think `if CPU_HAS_FPU` is not mandatory then. If you want `if` then it should probably be `select FPU if CPU_HAS_FPU`

Reply:
Removed imply dependency. I wanted to avoid using select to prevent us from forcing FPU to be enabled if available.
FPU speeds up calculations, but it's not mandatory.